### ISSUE TITLE
Program crash

### ISSUE THREAD

**Issue by gumpjt** [looterz/grimd#20]
After running for a while, it will crash. I check the source code but can't find the reason. Here is the log info:

```
panic: not reached 
   goroutine 859142 [running]:
   panic(0x83f320, 0xc4211353a0)
           /go-root/src/runtime/panic.go:500 +0x1a1
 github.com/miekg/dns.(*response).Write(0xc420065480, 0xc421078300, 0x25, 0x26, 0x0, 0x0, 0x7f7cc19c9000)
           /my-project-path/src/.vendor/src/github.com/miekg/dns/server.go:702 +0x325
   github.com/miekg/dns.(*response).WriteMsg(0xc420065480, 0xc421949200, 0x1, 0xc420d0baa0)
          /my-project-path/src/.vendor/src/github.com/miekg/dns/server.go:677 +0x9c
  github.com/miekg/dns.HandleFailed(0xade740, 0xc420065480, 0xc42035f0e0)
          /my-project-path/src/.vendor/src/github.com/miekg/dns/server.go:92 +0x8b
  main.(*DNSHandler).do(0xc420129380, 0x8d5330, 0x3, 0xade740, 0xc420065480, 0xc42035f0e0)
          /my-project-path/src/handler.go:223 +0x11db
  created by main.(*DNSHandler).DoTCP
         /my-project-path/src/handler.go:267 +0x72
```
My go version : go version go1.7.4 linux/amd64.

**Comment by photostorm** [comment]
Hi gumpjt, I pretty sure I know what cause that error to happen. It is something I fixed for the normal reply, but for HandledFailed function bypass my fixed. This is a case that happens when a TCP connection comes in and later get interrupted causing the writer to go nil. I can work on fixed for this.

**Comment by looterz** [comment]
Should be fixed by #21
